### PR TITLE
Add Endpoint for Viewing Lesson Completion Stats

### PR DESCRIPTION
**Pull request by KevinMulhern**
To keep an eye on lesson completion stats, we need access to lesson
completion data so we can make visulizations.

### CODE REVIEW

File: app/controllers/api/lesson_completions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      end
+    end
+
+    def days
+      params[:days].to_i
+    end
+
+    def authenticate
+      authenticate_or_request_with_http_basic do |username, password|
+        username == ENV.fetch('API_USERNAME')
```
Comment (arku) [inline app/controllers/api/lesson_completions_controller.rb]: @KevinMulhern Shouldn't this be `username == ENV.fetch('API_USERNAME') && password == ENV.fetch('API_PASSWORD')`? :thinking:

File: app/serializers/course_serializer.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+class CourseSerializer
+  attr_reader :course, :days
+  private :course, :days
```
Comment (arku) [inline app/serializers/course_serializer.rb]: @KevinMulhern I have never seen this before, can you help me understand what this does?

File: app/controllers/api/lesson_completions_controller.rb (hunk 1 above)
Comment (arku) [reply app/controllers/api/lesson_completions_controller.rb]: At its current state, the code only checks for the `password` and not the combination of `username` and `password`.

Comment (KevinMulhern) [reply app/controllers/api/lesson_completions_controller.rb]: So it is, well spotted.

File: app/serializers/course_serializer.rb (hunk 2 above)
Comment (KevinMulhern) [reply app/serializers/course_serializer.rb]: Sure, the `attr_reader` exposes course and days getter methods both internally and externally for this this object. We only want them to be accessible internally though, we don't want them to be part of the objects external API so we can make them private to achieve that.

**Comment by KevinMulhern** [comment]
@arku this is ready for re-review, when you get a moment please :)

**Comment by arku** [comment]
@KevinMulhern Sure, will take a look later today.

File: app/serializers/course_serializer.rb (hunk 2 above)
Comment (arku) [reply app/serializers/course_serializer.rb]: Ah, thanks for the explanation. 

I have always used something like this:

```ruby
class CourseSerializer
  # ...

  private

  attr_reader :course, :days
end

File: .env [hunk 3]
```diff
@@ -0,0 +1,5 @@
+GITHUB_API_TOKEN: 1234
```
Comment (arku) [inline .env]: Shouldn't we also add other keys like `MAILCHIMP_API_KEY`, `MAILCHIMP_LIST_ID` etc? :thinking:

Comment (KevinMulhern) [reply .env]: Ahh, good idea.

**Comment by KevinMulhern** [comment]
Thanks for all the reviewing you did on this PR @arku, I appreciate it 😄

**Pull request merged**